Attempting to use setuptools to allow installation of this package as dependency

giorgiosironi opened the pull request:
Main problem is the files are not in a package, so I'm not sure I can use the package as-is from another Python project (no problem in depending on the Git repo instead of PyPI). Thoughts?

Jenniferstrej commented:
Can't you try making it into a package? Adding a `__init__.py` for example? https://python-packaging.readthedocs.io/en/latest/minimal.html

giorgiosironi commented:
I'll do that, but it will require a subfolder for the scripts (don't know if there is any dependency on their path around).

Jenniferstrej commented:
I'm not sure what you mean by "don't know if there is any dependency on their path around". But by looking at the only .py files besides the tests, they don't depend on each other. econ_feeder needs environment variables to be set for the AWS credentials for example... But I don't see any issue with placing them in a subfolder. Do you think there are any issues @jhroot  ?

giorgiosironi commented:
e.g. cron setup somewhere pointing to them, aliases in people's `.bashrc` files, you never know :) I'll move them but after #4

giorgiosironi commented:
Rebased and updated by creating a package named `econtools` (convention IIUC is the Python package names are discouraged from using underscores and forbidden from using hyphens).

giorgiosironi commented on README.md:
```diff
@@ -51,7 +51,7 @@ Usage:
     econ_article_feeder.py [options] bucket_name workflow_starter_queue_name workflow_name
 
-    $ python econ_article_feeder.py -p elife-14721-vor-r1 -r 1  elife-production-final workflow-starter-queue IngestArticleZip
+    $ python econtools/econ_article_feeder.py -p elife-14721-vor-r1 -r 1  elife-production-final workflow-starter-queue IngestArticleZip
```
@gnott if you use this project, this is an upcoming change

giorgiosironi commented on setup.py:
```diff
@@ -0,0 +1,27 @@
+with open('requirements.txt') as f:
+    install_requires = f.read().splitlines()
+
+setup(name='econtools',
```
all these options:
<img src="http://s2.quickmeme.com/img/bc/bc75b955f736ac0cb29ee79aee4bf206103945487a1449dcecdc993d688f1cc1.jpg" width=200 />

seanwiseman replied:
I would change the version value https://github.com/elifesciences/econ-tools/blob/d91a41a2054ce5408ea6d46b3027a8ad3453be24/setup.py#L10

to be an imported value from the econtools/__init__.py: `__version__ = '0.0.1'`. It is the standard way, allows the version to be checked by users and keeps the `setup.py` up to date when the version gets bumped.

seanwiseman replied:
This email does not seem to work:
https://github.com/elifesciences/econ-tools/blob/d91a41a2054ce5408ea6d46b3027a8ad3453be24/setup.py#L18
I changed it to the tech team address in the last setup.py file I did:
`maintainer_email='tech-team@elifesciences.org',`

seanwiseman replied:
Is this going on PyPi?
